src/common/runtime/cmdline.ts

```diff
@@ -54,7 +67,33 @@ if (queries.length === 0) {
   try {
     const loader = new DefaultTestFileLoader();
     assert(queries.length === 1, 'currently, there must be exactly one query on the cmd line');
-    const testcases = await loader.loadCases(parseQuery(queries[0]));
+    const testcaseQuery = parseQuery(queries[0]);
+    const testcases = await loader.loadCases(testcaseQuery);
+
+    const expectations: TestQueryWithExpectation[] = [];
+    if (typeof loadWebGPUExpectations !== 'undefined') {
+      (await loadWebGPUExpectations).forEach(entry => {
+        const query = parseQuery(entry.query);
+
+        if (compareQueries(testcaseQuery, query) === Ordering.Unordered) {
+          return;
+        }
+
+        switch (entry.expectation) {
+          case 'pass':
+          case 'skip':
+          case 'fail':
+            break;
+          default:
+            unreachable();
```

Comment:
nit: Probably best to have a message here since this isn't an internal error.

optional nit: `assert(['pass', 'skip', 'fail'].includes(entry.expectation), '...');`

Reply:
I kept the switch statement. The other way doesn't narrow the type so it doesn't infer correctly.